packages/background/src/controllers/NetworkController.ts

```diff
@@ -73,12 +73,14 @@ export default class NetworkController extends BaseController<NetworkControllerS
         this._updateProviderNetworkStatus();
 
         // Set the error handler for the provider to check for network status
-        this.provider.on('error', this._updateProviderNetworkStatus);
+        this.provider.on('error', (err) => {
+            return this._updateProviderNetworkStatus({ err });
+        });
 
         // Periodical checks when provider is down
-        setInterval(() => {
+        setInterval(async () => {
             if (!this.getState().isProviderNetworkOnline) {
-                this._updateProviderNetworkStatus();
+                await this._updateProviderNetworkStatus({ timeout: 3000 });
```

Comment:
shouldn't we move this timeout value to a constant? could be potentially included in the remote config repo along with other network settings, so maybe it makes sense to have it on a constant so we don't forget about it

Reply:
This code has changed due to some features lacks I found in the `ethers.provider` implementation. 

We use the `poll` method to query the provider's status where we specify a 10seconds of timeout. However, this a virtual timeout since the library doesn't actually cancel the request, it only rejects the promise.
That being said, In 30 seconds of failure scenario we: 
- Check the provider's status => 10 seconds (1st request)
- Get the response and sleep => 5 seconds = 15 seconds
- Check the provider's status => 10 seconds (2nd request)  = 25 seconds 
- Get the response and sleep => 5 seconds = 30 seconds
- Check the provider's status => 10 seconds (3rd request) = 40 seconds
- Get the response and sleep => 5 seconds = 45 seconds
- Check the provider's status => 10 seconds (4th request) = 55 seconds
- Get the response and sleep => 5 seconds = 1 minute => 1st request is timeouted by the browser. 

In summary, in 1 minute we will have a max of 4 active request vs 20 request we used to have.